.github/workflows/deployTraining.yml

```diff
@@ -48,11 +48,18 @@ jobs:
         run: make init-${{ env.DEPLOY_ENV }}
       - name: Azure deploy Api
         run: make deploy-${{ env.DEPLOY_ENV }}-api
-      - name: Wait for staging deploy to complete
-        timeout-minutes: 10
-        run: make wait-for-${{ env.DEPLOY_ENV }}
+      - name: Wait for correct commit to be deployed in staging slot
+        timeout-minutes: 5
+        run: make wait-for-${{ env.DEPLOY_ENV }}-slot-commit
+      - name: Wait for staging deploy to be ready
+        timeout-minutes: 1
+        run: make wait-for-${{ env.DEPLOY_ENV }}-slot-readiness
       - name: Promote staging to production
         run: make promote-${{ env.DEPLOY_ENV }}
+      - name: Check for production app readiness
+        env:
+          CURL_TIMEOUT: 5 # 1 second seems to be too tight for the initial launch
```

Comment:
"Initial launch" meaning there's plans to tweak this later? Otherwise, we should just make the default 5 in the Makefile.

Reply:
The initial launch of the application—in general 1 second is plenty, but as the app is warming up it seems to sometimes take a little longer. I threw five here just to make sure I didn't get a spurious failure during testing, but didn't think about it very hard, since I figured that we would talk about it later (then I forgot to mention it in the notes, so thanks for calling this out!).

I think setting the time once in the file and removing the overrides is probably but not necessarily right, since the two goals are:
* never do ridiculous waits because the app gateway is holding onto the connection trying to get a server ready to respond to it
* never fail a simple check because the app server is a little logy at startup time

I initially set the cutoff to the most aggressive value possible because I was focused on the first point, but it turns out that the first point isn't really all that useful (we can still get spurious successes if we keep trying long enough), so I think it's fine to bump it up some globally to handle the second point. I think 5 seconds is still a bit much, so my reflex is to make it globally 2 seconds and eliminate the override here. Do you think that's likely enough, or should I just get over it and make it 5 globally?

The other option is to have separate connect-timeout and max-time values, but that seems overly complicated.

Reply:
I think 2 seconds is probably fine. I'd rather err on the side of taking an extra 1-2 seconds to avoid churn but like you said I don't really know if there's any meaningful difference between 2 and 5 seconds here.